### DISCUSSION TITLE
SpacePy irbempylib in Centos7

### DISCUSSION THREAD

**Discussion by dshanaberger** [spacepy/spacepy#457] [Support and Q&A] [answered]
I am working with irbempy library within SpacePy on Centos7. I am coming across a traceback error when calling on irbempylib. 

Traceback (most recent call last):
  File "********.py", line 26, in <module>
    from spacepy import irbempy as ib
  File "/usr/local/lib/python3.6/site-packages/spacepy/irbempy/__init__.py", line 71, in <module>
    from .irbempy import *
  File "/usr/local/lib/python3.6/site-packages/spacepy/irbempy/irbempy.py", line 31, in <module>
    from . import irbempylib as oplib
ImportError: cannot import name 'irbempylib'

This may be the result of an improper build of spacepy... although it seems that spacepy build is working.

**Comment by jtniehof** [comment, marked as answer]
That traceback indicates that the irbempy component didn't install when you installed SpacePy. Usually this comes from not having a Fortran compiler installed; we've also got some recent commits (not in the latest release version) that improve compatibility with more recent compilers.

If irbempy fails to compile, the installer will proceed on the assumption that the user might not care. (This is also true of some other optional components.) In that case, the last line of the install will be: "setup produced the following warnings. Some functionality may be missing." and then a line of detailed warnings. So the way for us to debug this would be to have 1) how you installed SpacePy 2) a full typescript of the install process so we can find those warnings (and anything that lead up to it.)

EDIT: As in #604, relevant information on debugging this sort of issue can be found in the installation docs on [troubleshooting irbempy](https://spacepy.github.io/install.html#irbempy) and [troubleshooting pip](https://spacepy.github.io/install.html#pip-failures), including capturing the full pip output. As outlined there, I'll be closing this discussion since much has changed on that front...opening discussions for issues is still welcome.
